docs/machine-learning/how-to-guides/inspect-intermediate-data-ml-net.md

```diff
@@ -1,75 +1,151 @@
+> [!WARNING]
+> [`CreateEnumerable`](xref:Microsoft.ML.DataOperationsCatalog.CreateEnumerable*) will load the entire [`IDataView`](xref:Microsoft.ML.IDataView) into memory which may affect performance.
 
-[Example file](https://github.com/dotnet/machinelearning/tree/master/test/data/adult.tiny.with-schema.txt):
+```csharp
+// Create an IEnumerable from IDataView
+HousingData[] housingDataEnumerable =
+    mlContext.Data.CreateEnumerable<HousingData>(data, reuseRowObject: false).ToArray();
 
-<!-- markdownlint-disable MD010 -->
+// Get first element of IEnumerable
+HousingData firstRow = housingDataEnumerable[0];
 ```
-Label	Workclass	education	marital-status
-0	Private	11th	Never-married
-0	Private	HS-grad	Married-civ-spouse
-1	Local-gov	Assoc-acdm	Married-civ-spouse
-1	Private	Some-college	Married-civ-spouse
 
+## Inspect values in a single column
+
+At any point in the process, values in a single column of an [`IDataView`](xref:Microsoft.ML.IDataView) can be accessed using the [`GetColumn`](xref:Microsoft.ML.Data.ColumnCursorExtensions.GetColumn*) method. The [`GetColumn`](xref:Microsoft.ML.Data.ColumnCursorExtensions.GetColumn*) method returns all of the values in a single column as an [`IEnumerable`](xref:System.Collections.Generic.IEnumerable`1)
+
+```csharp
+IEnumerable<float> sizeColumn = data.GetColumn<float>("Size").ToList();
 ```
-<!-- markdownlint-enable MD010 -->
 
-Our class is defined as follows:
+## Inspect values one row at a time
+
+[`IDataView`](xref:Microsoft.ML.IDataView) is lazily evaluated. To iterate over the rows, create a [`DataViewRowCursor`](xref:Microsoft.ML.DataViewRowCursor) by using the [`GetRowCursor`](xref:Microsoft.ML.IDataView.GetRowCursor*) method and passing in the [DataViewSchema](xref:Microsoft.ML.DataViewSchema) of your [`IDataView`](xref:Microsoft.ML.IDataView) as a parameter. Then, to iterate over rows, use the [`MoveNext`](xref:Microsoft.ML.DataViewRowCursor.MoveNext*) cursor method. 
 
 ```csharp
-public class InspectedRow
+
+DataViewSchema columns = data.Schema;
+
+using (DataViewRowCursor cursor = data.GetRowCursor(columns))
 {
-    [LoadColumn(0)]
-    public bool IsOver50K { get; set; }
-    [LoadColumn(1)]
-    public string WorkClass { get; set; }
-    [LoadColumn(2)]
-    public string Education { get; set; }
-    [LoadColumn(3)]
-    public string MaritalStatus { get; set; }
+    float size = default;
+    VBuffer<float> historicalPrices = default;
+    float currentPrice = default;
+
+    while (cursor.MoveNext())
+    {
+        cursor.GetGetter<float>(columns[0]).Invoke(ref size);
+        cursor.GetGetter<VBuffer<float>>(columns[1]).Invoke(ref historicalPrices);
+        cursor.GetGetter<float>(columns[2]).Invoke(ref currentPrice);
+    }
 }
 ```
 
+> [!IMPORTANT]
+> For performance purposes, vectors in ML.NET use [`VBuffer`](xref:Microsoft.ML.Data.VBuffer`1) instead of native collection types (i.e. Vector,float[]). 
+
+## Preview result of pre-processing or training on a subset of the data
```

Comment:
This looks great!

Can we add a caution to say not to use Preview in production code?

Reply:
#resolved